app/assets/javascripts/oxalis/view/right-menu/meshes_view.js

```diff
@@ -0,0 +1,182 @@
+// @flow
+
+import { Button, Upload, Checkbox, Icon, Input, Modal, Spin } from "antd";
+import { connect } from "react-redux";
+import React from "react";
+
+import type { MeshMetaData, RemoteMeshMetaData } from "admin/api_flow_types";
+import type { OxalisState } from "oxalis/store";
+import type { Vector3 } from "oxalis/constants";
+import { Vector3Input } from "libs/vector_input";
+import {
+  createMeshFromBufferAction,
+  deleteMeshAction,
+  updateLocalMeshMetaDataAction,
+  updateRemoteMeshMetaDataAction,
+} from "oxalis/model/actions/annotation_actions";
+import { readFileAsArrayBuffer } from "libs/read_file";
+import { setImportingMeshStateAction } from "oxalis/model/actions/ui_actions";
+import ButtonComponent from "oxalis/view/components/button_component";
+
+const ButtonGroup = Button.Group;
+
+const mapStateToProps = (state: OxalisState) => ({
+  meshes: state.tracing != null ? state.tracing.meshes : [],
+  isImporting: state.uiInformation.isImportingMesh,
+});
+
+const mapDispatchToProps = (dispatch: Dispatch<*>) => ({
+  updateRemoteMeshMetadata(id: string, meshMetaData: $Shape<RemoteMeshMetaData>) {
+    dispatch(updateRemoteMeshMetaDataAction(id, meshMetaData));
+  },
+  deleteMesh(id: string) {
+    dispatch(deleteMeshAction(id));
+  },
+  onChangeVisibility(mesh: MeshMetaData, isVisible: boolean) {
+    dispatch(updateLocalMeshMetaDataAction(mesh.id, { isVisible }));
+  },
+  async onStlUpload(info) {
+    dispatch(setImportingMeshStateAction(true));
+    const buffer = await readFileAsArrayBuffer(info.file);
+    dispatch(createMeshFromBufferAction(info.file.name, buffer));
+  },
+});
+
+type StateProps = {|
+  meshes: Array<MeshMetaData>,
+  isImporting: boolean,
+|};
+type DispatchProps = $Call<typeof mapDispatchToProps>;
+
+type Props = { ...DispatchProps, ...StateProps };
+
+const getCheckboxStyle = isLoaded =>
+  isLoaded
+    ? {}
+    : {
+        fontStyle: "italic",
+        color: "#989898",
+      };
+
+class EditMeshModal extends React.PureComponent<
```

Comment:
Maybe move this auxiliary class further up. It is a little confusing that there is a Props type and `map{Dispatch/State}ToProps`above this class, but those are actually for the second class in the file.